eh_win32_msvc.cc

```diff
@@ -203,10 +210,52 @@ OBJC_PUBLIC extern "C" void objc_exception_throw(id object)
 	__builtin_unreachable();
 }
 
-
 OBJC_PUBLIC extern "C" void objc_exception_rethrow(void* exc)
 {
 	_CxxThrowException(nullptr, nullptr);
 	__builtin_unreachable();
 }
 
+// rebase_and_cast adds a constant offset to a U value, converting it into a T
+template <typename T, typename U>
+static std::add_const_t<std::decay_t<T>> rebase_and_cast(size_t base, U value) {
```

Comment:
Please don't use `size_t` for values that carry pointer provenance.  `intptr_t` exists for that purpose.